src/components/screens/main-screen.tsx

```diff
@@ -127,54 +147,129 @@ const MainScreen: React.FC = () => {
   const renderCodeSearch = () => {
     const courseOptions = courseAliases.map(alias => ({ label: alias.code, name: `${alias.code} ${alias.name}` }));
 
+    // cut name after the first , or (, then add ... if name is too long
+    courseOptions.forEach(option => {
+      let commaIndex = option.name.indexOf(",");
+      commaIndex === -1 && (commaIndex = option.name.length);
+      let bracketIndex = option.name.indexOf("(");
+      bracketIndex === -1 && (bracketIndex = option.name.length);
+      const cutIndex = Math.min(commaIndex, bracketIndex);
+      option.name = option.name.substring(0, cutIndex);
+      if (option.name.length > 49) {
+        option.name = `${option.name.substring(0, 49)}...`;
+      }
+    });
```

Comment:
Why do we need to do this?

Reply:
Before, the items in the drop-down menu were almost all cut off because they all had too long text, now they are cropped to the right length

Reply:
I'll add comments